keps/sig-testing/2420-reducing-kubernetes-build-maintenance/README.md

```diff
@@ -0,0 +1,671 @@
+-->
+
+Kubernetes currently maintains multiple build systems, an ongoing burden and a source of contributor friction and confusion. Much has changed since Bazel was first introduced as an additional build system, upon re-evaluating the project it is clear that we should dedupe this. More details on why can be found in [Motivation](#motivation) and [Drawbacks](#drawbacks).
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Remove the toil of maintaining multiple build-systems from the Kubernetes repo maintainers
+- Eliminate the friction of generating `BUILD` files from Kubernetes contributors using Go natively
+- Simplify Golang upgrades for SIG Release
+- Remove qualification gaps caused by duplicate-but-slightly-different binary builds
+- Remove testing gaps caused by duplicate-but-slightly-different test-invocation methods
+- Empower broader community maintenance of tests by converging on Golang testing standards
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Support importing github.com/kubernetes/kubernetes as a library
+  - This has never been supported, and is orthogonal to the decision of maintaining duplicate build systems
+  - Code under kubernetes/kubernetes/src/staging/k8s.io/foo is intended to be imported as k8s.io/foo from the staged copies at github.com/kubernetes/foo; anything else is internal-only and not supported for import, bazel or otherwise.
+- Removing Bazel from sub-projects other than the core repo
+  - Bazel comes with tradeoffs, each subproject can make its own decision in this regard. This KEP **only** covers the main Kubernetes repository, and nothing else.
+- Improving the previously existing make build
+  - We should strongly consider improving the implementation and behavior of this build in the future, but this is largely orthogonal to whether we should consider maintaining two build systems.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+For the development (main/master) branch only, **NOT** the existing release branches:
```

Comment:
👍🏾

Reply:
considered, and rejected for initial implementation, but something we'll keep an eye on.